src/utils/objects/map.js

```diff
@@ -0,0 +1,34 @@
+import isPlainObject from 'lodash.isplainobject';
+
+function map(input, callback, options) {
```

Comment:
I'm assuming you put `options` as the last argument since you may not have `options` to pass down. I'm not sure how other libraries do this but it does feel a little odd to me to not have the iterator function last but I'm not sure I can think of a better solution where the iterator is always last

Reply:
The native APIs for stuff like `map` and `forEach` do this (optional arguments last, specifically the `thisArg`), as does Underscore. It's pretty common.

Reply:
What about doing something like the following?
```
function map(input, options, callback) {
  callback = callback || options;
  options = typeof options === 'function'
    ? { deep: false }
    : { deep: false, ...options }

  // the rest of the function here
}
```

Reply:
Also should there be any JSDocs for these private functions?

Reply:
1. All evidence I have seen (how native array iterators and Underscore handle optional args, [AirBnb style guide on default parameters](https://github.com/airbnb/javascript#functions--defaults-last)) points to your assumption (the iterator needs to be last in this case) being wrong. I think it all stems back to function overloading not being something built into JS. Are you aware of any counter-examples off the top of your head? I know I've seen them before. I'd really rather avoid doing two reassignments to just move the optional argument.

1. Yes, there should be JSDocs for these. Whoops.